src/main/JayZ.ts

```diff
@@ -25,69 +31,99 @@ export class JayZ {
         ? config.encryptor
         : new LibsodiumEncryptor()
     this.maxUsesPerDataKey = config.maxUsesPerDataKey || 1
+    this.currentDataKeyUsesRemaining = this.maxUsesPerDataKey
   }
 
-  async encryptItem<T, K extends keyof T>(
-    item: T,
-    fieldsToEncrypt: K[]
-  ): Promise<EncryptedJayZItem<T, K>> {
-    const { dataKey, encryptedDataKey } = await this.getDataKey()
-    const { encryptedItem, nonce } = await this.encryptor.encrypt({
-      item,
-      fieldsToEncrypt,
-      dataKey
+  async encryptItems<T, K extends keyof T>(
+    itemsToEncrypt: EncryptItemRequest<T, K>[]
+  ): Promise<EncryptedJayZItem<T, K>[]> {
+    if (itemsToEncrypt.length === 0) {
+      return []
+    }
+
+    const itemsWithKey = await this.zipItemsWithDataKey(itemsToEncrypt)
```

Comment:
This method now ensures that if we request to encrypt N items in the same event loop tick, that we generate the expected number of data keys implied by the `maxUsesPerDataKey` setting